demo/lego_official_projects/common.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+
+import logging
+import sys
+import time
+from ev3dev.auto import *
+
+log = logging.getLogger(__name__)
+
+
+class Tank(object):
+
+    def __init__(self, left_motor, right_motor):
+
+        for motor in (left_motor, right_motor):
+            if motor not in OUTPUTS:
+                log.error("%s in an invalid output, choices are %s" % (motor, ''.join(OUTPUTS)))
+                sys.exit(1)
+
+        self.left_motor = LargeMotor(left_motor)
+        self.right_motor = LargeMotor(right_motor)
+
+        for x in (self.left_motor, self.right_motor):
+            if not x.connected:
+                log.error("%s is not connected" % x)
+                sys.exit(1)
+
+        self.left_motor.reset()
+        self.right_motor.reset()
+        self.speed = 90
```

Comment:
The name `speed` is confusing, because (1) it is used to control `duty_cycle_sp`, and (2) there is actually `Motor.speed` attribute which is read-only and returns current speed. I suggest renaming this to `duty_cycle_sp`.